### PR TITLE
[TM] Update instructions to create custom exercise data 

### PR DESCRIPTION
**Pull request by DelazJ**
Because the native tool to add OSM vector file has been removed, we need to rely on a third-party plugin ie QuickOSM
Add new training projects folder to keep track of exercises... 

@Gustry could you please give a look whether i use the plugin in its optimal way and am right in what i wrote, please? Thanks.

### CODE REVIEW

File: source/docs/training_manual/foreword/preparing_data.rst [hunk 1]
```diff
@@ -37,127 +37,112 @@ should have a good mix of urban and rural areas, containing roads of differing
...
+the process for the other layers.
+Each resulting data set should be saved in the :file:`shapefiles` directory in
+your new :file:`exercise_data` directory (i.e. "water", "landuse").
+To be compatible with the course data, you would need to rename:
+
+* ``natural_water`` into ``water``;
+* ``waterway_river`` into ``rivers``;
+* ``building`` into ``buildings``;
+* ``place`` into ``places``;
+* ``highway`` into ``roads``.
```
Comment (DelazJ) [inline source/docs/training_manual/foreword/preparing_data.rst]: @ghtmtt i wonder if we should ask to rename (moving to plural most of the time) here or instead rename our files in data to match OSM tags. We would then need to rename `waterway_river` and `natural_water`

**Comment by DelazJ** [comment]
The fail is fixed in #3053

**Comment by Gustry** [comment]
Sorry, I missed the notification.
I will review it tomorrow.

I have just released QuickOSM 1.9.0 which improves the UI. Sorry it needs to update screenshots. I can do it later. It's about the about extent selector in the Quick Query panel.

BTW, @ghtmtt the download all keys is now the default. You made this update for the QGIS Training manual.

**Comment by Gustry** [comment]
Sorry, I missed the notification.
I will review it tomorrow.

I have just released QuickOSM 1.9.0 which improves the UI. Sorry it needs to update screenshots. I can do it later. It's about the about extent selector in the Quick Query panel.

BTW, @ghtmtt the download all keys is now the default. You made this update for the QGIS Training manual.

**Comment by Gustry** [comment]
Sorry, I missed the notification.
I will review it tomorrow.

I have just released QuickOSM 1.9.0 which improves the UI. Sorry it needs to update screenshots. I can do it later. It's about the about extent selector in the Quick Query panel.

BTW, @ghtmtt the download all keys is now the default. You made this update for the QGIS Training manual.

**Comment by Gustry** [comment]
Sorry, github is having issues since yesterday evening. I will redo my review

File: source/docs/training_manual/foreword/preparing_data.rst [hunk 2]
```diff
@@ -32,127 +32,121 @@ should have a good mix of urban and rural areas, containing roads of differing
...
+
+#. Go to :menuselection:`Plugins --> Manage/Install Plugins...`
+#. In the :guilabel:`All` tab, type ``QuickOSM`` in the search box;
+#. Select the QuickOSM plugin, press **[Install plugin]** and then **[Close]**
+   the dialog;
+
+   .. image:: img/quickosm_plugin_download.png
+      :align: center
+
+#. Execute the new plugin from :menuselection:`Vector --> Quick OSM --> QuickOSM`
```
Comment (Gustry) [inline source/docs/training_manual/foreword/preparing_data.rst]: > Quick OSM

QuickOSM, it has been fixed

File: source/docs/training_manual/foreword/preparing_data.rst [hunk 3]
```diff
@@ -32,127 +32,121 @@ should have a good mix of urban and rural areas, containing roads of differing
...
+#. In the :guilabel:`Quick query` tab, check the |radioButtonOn| :guilabel:`Current
```
Comment (Gustry) [inline source/docs/training_manual/foreword/preparing_data.rst]: There is no more a radio button. I updated to a checkbox

File: source/docs/training_manual/foreword/preparing_data.rst [hunk 4]
```diff
@@ -32,127 +32,121 @@ should have a good mix of urban and rural areas, containing roads of differing
...
-.. image:: img/building_query_builder.png
-   :align: center
+   .. note:: If you need to replace some particular properties (CRS, extent,
+    fields...) of the temporary layer, use instead the :menuselection:`Export
+    --> Save Features as...` contextual menu, and ensure the :guilabel:`Add
+    saved file to map` option is checked. A new layer is added in such case.
 
-* Click :guilabel:`Ok` and you'll see that the layer elements which are not
-  buildings have been removed from the map.
+#. Make sure the format is :guilabel:`ESRI Shapefile`;
```
Comment (Gustry) [inline source/docs/training_manual/foreword/preparing_data.rst]: We should switch to a more OGC format oneday ;-)

**Comment by ghtmtt** [comment]
@Gustry thanks for the review. Just one question about the QuickOSM (that can actually be my fault with the *download all* option): I see that sometimes there are overlapped polygons (whole polygon of the region covering all the buildings). I don't remember how the old core OSM plugin worked on this...

File: source/docs/training_manual/foreword/preparing_data.rst (hunk 3 above)
Comment (DelazJ) [reply source/docs/training_manual/foreword/preparing_data.rst]: You meant a combobox, right? ;)

File: source/docs/training_manual/foreword/preparing_data.rst (hunk 4 above)
Comment (DelazJ) [reply source/docs/training_manual/foreword/preparing_data.rst]: yeah we should. I think it should be the target for 3.10 docs. I don't find gpkg in QGIS stable enough (got too much issues with fid and had to workaround, so not yet a good format for training manual imho)

**Comment by DelazJ** [comment]
> I think screenshots will need an update with the latest release (on QGIS 3.3 only). Sorry for that.

No problem. I'll update description and screenshot.
> I guess I won't change the UI anymore, except if someone brings something better ;-)

👍

**Comment by DelazJ** [comment]
And thanks for the review

File: source/docs/training_manual/foreword/preparing_data.rst (hunk 3 above)
Comment (Gustry) [reply source/docs/training_manual/foreword/preparing_data.rst]: Ah yes !

File: source/docs/training_manual/foreword/preparing_data.rst (hunk 4 above)
Comment (Gustry) [reply source/docs/training_manual/foreword/preparing_data.rst]: I agree. For analysis, I got this issue too.
It seems that a fix is coming: https://github.com/qgis/QGIS/pull/8240

**Comment by Gustry** [comment]
> I see that sometimes there are overlapped polygons (whole polygon of the region covering all the buildings)

@ghtmtt It's all OSM data within the extent. You expect big polygons to be at the back? I don't think I can do something. Did you try to use the 'sort by expression' algorithm and sort by area?

**Comment by ghtmtt** [comment]
@Gustry thanks for the hint ;)

File: source/docs/training_manual/foreword/preparing_data.rst [hunk 5]
```diff
@@ -32,127 +32,126 @@ should have a good mix of urban and rural areas, containing roads of differing
...
+   directory.
+   
+   .. note:: For convenience, you'll also need to rename each layer:
+    double-click the layer to open its :menuselection:`Layer Properties -->
+    Source` tab and update the :guilabel:`Layer name` property accordingly.
+    
 
-Each resulting data set should be saved in the "epsg4326" directory in your new
-:kbd:`exercise_data` directory (i.e. "water", "landuse").
+The last step is to create a spatiaLite file from the :file:`landuse` layer to
```
Comment (ghtmtt) [inline source/docs/training_manual/foreword/preparing_data.rst]: I'm OK with SpatiaLite of course, but (sorry if I missed something recently), any reason to don't put all the data in a geopackage?

**Comment by ghtmtt** [comment]
do you mind to add an anchor like `_preparing_data` here: https://github.com/qgis/QGIS-Documentation/pull/3054/files#diff-b3782cda59b6d47fac5b84534634f413R25? 

+1 to merge anyway ;)

File: source/docs/training_manual/foreword/preparing_data.rst (hunk 5 above)
Comment (DelazJ) [reply source/docs/training_manual/foreword/preparing_data.rst]: Has always bee a SpatiaLite. For consistency and prepare gpkg usage, I'll with a gpkg format. In another PR.

**Comment by DelazJ** [comment]
done: `tm_preparing_data`

**Comment by ghtmtt** [comment]
> done: tm_preparing_data

awesome thanks. 

go and merge! ;)

**Pull request merged**